src/mp4.cpp

```diff
@@ -143,6 +143,30 @@ MP4FileHandle MP4ReadProvider( const char* fileName, const MP4FileProvider* file
     return MP4_INVALID_FILE_HANDLE;
 }
 
+///////////////////////////////////////////////////////////////////////////////
+
+void MP4SetShouldParseAtomCallback( MP4FileHandle hFile, ShouldParseAtomCallback cb )
+{
+   if (!MP4_IS_VALID_FILE_HANDLE(hFile))
```

Comment:
This looks whole function looks funky since it's a pattern I'm not familiar with. But I assume it's a pattern inside of mp4v2 to make their C interface work with the underlying C++ code?

Reply:
It seems like it. If you browse through the rest of the file, you'll see this is just following the existing template.